targets/unity-v2/Testing/Tests/Entity/EntityApiTests.cs

```diff
@@ -141,6 +145,194 @@ private void GetObjectCallback2(GetObjectsResponse result)
+        /// </summary>
+
+        [UUnitTest]
+        public void PutApi(UUnitTestContext testContext)
+        {
+            var loginRequest = new ClientModels.LoginWithCustomIDRequest
+            {
+                CustomId = PlayFabSettings.BuildIdentifier,
+                CreateAccount = true,
+                LoginTitlePlayerAccountEntity = true,
+            };
+
+            PlayFabClientAPI.LoginWithCustomID(loginRequest, PlayFabUUnitUtils.ApiActionWrapper<ClientModels.LoginResult>(testContext, LoginCallbackPutTest), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+        }
+        private void LoginCallbackPutTest(ClientModels.LoginResult result)
+        {
+            var testContext = (UUnitTestContext)result.CustomData;
+
+            if (result.EntityToken != null)
+            {
+                PlayFab.EntityModels.EntityTypes entityType = (PlayFab.EntityModels.EntityTypes)(int)result.EntityToken.Entity.Type; // possible loss of data 
+
+                _entityKey = new PlayFab.EntityModels.EntityKey
+                {
+                    Id = result.EntityToken.Entity.Id,
+                    Type = entityType,
+                    TypeString = result.EntityToken.Entity.TypeString
+                };
+
+                LoadFiles(testContext);
+            }
+            else
+            {
+                testContext.EndTest(UUnitFinishState.FAILED, "Entity Token is null!");
+            }
+        }
+        private void LoadFiles(UUnitTestContext testContext)
+        {
+            var request = new GetFilesRequest
+            {
+                Entity = _entityKey
+            };
+
+            PlayFabEntityAPI.GetFiles(request, PlayFabUUnitUtils.ApiActionWrapper<EntityModels.GetFilesResponse>(testContext, OnGetFilesInfo), PlayFabUUnitUtils.ApiActionWrapper<PlayFabError>(testContext, SharedErrorCallback), testContext);
+        }
+        void OnGetFilesInfo(GetFilesResponse result)
+        {
+            var testContext = (UUnitTestContext)result.CustomData;
+            bool testFileFound = false;
+            GetFileMetadata fileMetaData = new GetFileMetadata();
+
+            foreach (var eachFilePair in result.Metadata)
+            {
+                if (eachFilePair.Key.Equals(TEST_FILE_NAME))
+                {
+                    testFileFound = true;
+
+                    fileMetaData = eachFilePair.Value;
+                    break; // this test only support one file
+                }
+            }
+
+            if (!testFileFound)
+            {
+                UploadFile(TEST_FILE_NAME, testContext);
+            }
+            else
+            {
+                GetActualFile(fileMetaData, testContext);
+            }
+        }
+        void GetActualFile(GetFileMetadata fileData, UUnitTestContext testContext)
+        {
+            PlayFabHttp.SimpleGetCall(fileData.DownloadUrl,
+                result =>
```

Comment:
These simple callbacks are almost certainly the right way for customers to write this.
But unfortunately, if they fail in some way, they don't communicate the test failure back to the testContext.
For example, add:
```throw new Exception("do not commit this line");```
inside of one of these little lambda callbacks.
When you do, the test won't report that error, it will just time out mysteriously.

That's why the ApiActionWrapper exists for PfApi callbacks.
I'd suggest making another ApiActionWrapper in the same place, that wraps calls meant for SimpleGet/SimplePut.
I know it seems overly complex for testing the success case, but we also need to make sure this test captures failure cases and reports them in a useful way.

Reply:
(I'm not gonna repeat this comment in all the relevant places, but essentially EVERY lambda callback needs to be wrapped such that if it throws an exception, that exception gets properly displayed as a test failure, and not a test timeout)

Reply:
Created 2 more wrappers